src/ynEIGEN/TokenStakingNode.sol

```diff
@@ -124,76 +195,108 @@ contract TokenStakingNode is ITokenStakingNode, Initializable, ReentrancyGuardUp
 
+    /**
+     * @notice Returns the queued shares and withdrawn balance for a specific strategy and asset.
+     * @dev The queued shares are the sum of the legacy queued shares and the post slashing queued shares.
+     */
     function getQueuedSharesAndWithdrawn(IStrategy _strategy, IERC20 _asset) external view returns (uint256, uint256) {
-        return (queuedShares[_strategy], withdrawn[_asset]);
+        return (legacyQueuedShares[_strategy] + queuedShares[_strategy], withdrawn[_asset]);
     }
 
     /**
      * @notice Queues withdrawals for a specific strategy
      * @param _strategy The strategy from which to withdraw
-     * @param _shares The number of shares to withdraw
+     * @param _depositShares The number of deposit shares to withdraw. The amount of withdrawn shares on completion might be lower due to slashing.
      * @return _fullWithdrawalRoots An array of withdrawal roots generated by the queueWithdrawals operation
      */
-    function queueWithdrawals(IStrategy _strategy, uint256 _shares)
+    function queueWithdrawals(IStrategy _strategy, uint256 _depositShares)
         external
         onlyTokenStakingNodesWithdrawer
         onlyWhenSynchronized
         returns (bytes32[] memory _fullWithdrawalRoots)
     {
         IStrategy[] memory _strategiesArray = new IStrategy[](1);
         _strategiesArray[0] = _strategy;
-        uint256[] memory _sharesArray = new uint256[](1);
-        _sharesArray[0] = _shares;
+        uint256[] memory _depositSharesArray = new uint256[](1);
+        _depositSharesArray[0] = _depositShares;
         IDelegationManagerTypes.QueuedWithdrawalParams[] memory _params = new IDelegationManagerTypes.QueuedWithdrawalParams[](1);
         _params[0] = IDelegationManagerTypes.QueuedWithdrawalParams({
             strategies: _strategiesArray,
-            depositShares: _sharesArray,
-            __deprecated_withdrawer: address(this)
+            depositShares: _depositSharesArray,
+            __deprecated_withdrawer: address(0)
         });
 
-        queuedShares[_strategy] += _shares;
+        DelegationManagerStorage _delegationManager = DelegationManagerStorage(address(tokenStakingNodesManager.delegationManager()));
+
+        // `onlyWhenSynchronized` is used so we can assume that the operator is the same as the one in the DelegationManager.
+        address _operator = delegatedTo;
+        uint256 _withdrawableShares;
+        bytes32 _withdrawalRoot;
+
+        if (_operator == address(0)) {
+            _fullWithdrawalRoots = _delegationManager.queueWithdrawals(_params);
+            _withdrawalRoot = _fullWithdrawalRoots[0];
+            IDelegationManagerTypes.Withdrawal memory _queuedWithdrawal = _delegationManager.getQueuedWithdrawal(_withdrawalRoot);
+
+            // If the staker has not yet delegated to an operator, and given that this contract does not handle the beacon chain strategy.
```

Comment:
what if it was delegated in the past and then undelegated (_operator =0 case currently) , how does that impact this assumption?

is it still safe to assume `scaledShares` == `withdrawableShares` ?